agent/app/agent.go

```diff
@@ -164,9 +167,15 @@ func (agent *ecsAgent) doStart(containerChangeEventStream *eventstream.EventStre
 		log.Criticalf("Error creating state manager: %v", err)
 		return exitcodes.ExitTerminal
 	}
-	// TODO:
-	// 1) Feature gate ECS cgroup root init
-	// 2) Update mocks and unit tests
+
+	// Conditionally create '/ecs' cgroup root
+	if agent.cfg.TaskCPUMemLimit {
+		err = agent.resource.Init()
+		if err != nil {
+			log.Criticalf("Unable to setup '/ecs' cgroup: %v", err)
+			return exitcodes.ExitTerminal
```

Comment:
Can you add a comment here explaining why this should be treated as `Terminal` ?

Reply:
Sure